packages/xstate-store/src/atom.ts

```diff
@@ -86,16 +111,32 @@ export function createAtom<T>(
           e.stop();
         }
       };
+    },
+    _update(getValue?: () => T): boolean {
+      const prevSub = activeSub;
+      const compare = options?.compare ?? Object.is;
+      activeSub = atom as InternalReadonlyAtom<T>;
+      startTracking(atom as InternalReadonlyAtom<T>);
+      try {
+        const oldValue = atom._snapshot;
+        const read = (atom: Readable<any>) => atom.get();
+        const newValue = getValue ? getValue() : getter(read);
```

Comment:
this is a little bit tricky - does it mean a single implementation us reused by 2 quite different code paths? `getValue` is a dynamic updater~ wherear `getter` is somewhat static (configured when creating the atom).

This is fine but I think it deserves a code comment